### PR TITLE
Views

### PR DESCRIPTION
**Pull request by paulshaun-kdtk**
# Changes

- added views to list users
- added view to list posts
- added likes controller and route
- added comments controller and route
- implemented likes count, comment count and comments display

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -3,17 +3,24 @@
...
     <h2>Users</h2>
   </main>
 
+  <% @users_with_posts_count.each do |user| %>
+    <div>
+      <h2><%= link_to user.name, user %></h2>
+      <p>Posts count: <%= user.posts_count %></p>
+    </div>
+  <% end %>
+
```
Comment (VICTOR-CHUKWU) [inline app/views/users/index.html.erb]: - [ ] It is nice to keep to design guidelines, kindly match your UI to the given wireframe
in your design :point_down: 
Screenshot from 2023-11-25 11-42-42
given wireframe :point_down: 
Screenshot from 2023-04-21 17-06-41

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -7,20 +6,25 @@
...
+  <% @recent_posts.each do |post| %>
+    <div>
+      <h3><%= post.title %></h3>
+      <p><%= post.text %></p>
+    </div>
+  <% end %>
+    
+  <p><%= link_to "All Posts", user_posts_path(@user) %></p>
+  
+  <li><a href="/">back to home</a></li>
```
Comment (VICTOR-CHUKWU) [inline app/views/users/show.html.erb]: - [ ] It is nice to keep to design guidelines, kindly match your UI to the given wireframe
in your design :point_down: 
Screenshot from 2023-11-25 11-47-37
given wireframe :point_down: 
Screenshot from 2023-07-20 22-23-13

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -1,10 +1,18 @@
...
+  <h1><%= @user.name %>'s Posts</h1>
+
+  <% @posts.each do |post| %>
+    <div>
+      <h3><%= link_to post.title, post_path(post) %></h3>
+      <p><%= post.text %></p>
+    </div>
+  <% end %>
+  
+  <li><a href="/">back to home</a></li>
```
Comment (VICTOR-CHUKWU) [inline app/views/posts/index.html.erb]: - [ ] It is nice to keep to design guidelines, kindly match your UI to the given wireframe
in your design :point_down: 
Screenshot from 2023-11-25 11-50-08
given wireframe :point_down: 
Screenshot from 2023-05-11 17-37-59

File: app/views/posts/show.html.erb [hunk 4]
```diff
@@ -1,18 +1,27 @@
...
+  <p><strong>Comments:</strong> <%= @post.comments_count %></p>
+
+  <h2>Comments:</h2>
+  <% @post.ordered_comments.each do |comment| %>
+    <div>
+      <p><strong><%= comment.user.name %>:</strong> <%= comment.text %></p>
+    </div>
+  <% end %>
+
+  <p><%= link_to "Back to All Posts", user_posts_path(@post.author_id) %></p>
```
Comment (VICTOR-CHUKWU) [inline app/views/posts/show.html.erb]: - [ ] It is nice to keep to design guidelines, kindly match your UI to the given wireframe
in your design :point_down: 
Screenshot from 2023-11-25 11-51-58
given wireframe :point_down: 
Screenshot from 2023-09-01 21-07-50

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @paulshaun-kdtk  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by binhussen — APPROVED** [review]
Hi @paulshaun-kdtk , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉
IApproveSimonCowellGIF (6)

## Highlights
- Sticked Given Wireframe ✔️
- Well Stractured Code ✔️
- Used Git flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**